src/ui/StatusEditor.svelte

```diff
@@ -26,9 +56,13 @@
         const taskWithEditedStatusApplied = task.handleNewStatus(selectedStatus).pop();
 
         if (taskWithEditedStatusApplied) {
-            // Update the doneDate field, in case changing the status changed the value:
-            editableTask.doneDate = taskWithEditedStatusApplied.done.formatAsDate();
-            editableTask.cancelledDate = taskWithEditedStatusApplied.cancelled.formatAsDate();
+            setStatusRelatedDate('doneDate', selectedStatus.isCompleted(), taskWithEditedStatusApplied, 'done');
+            setStatusRelatedDate(
+                'cancelledDate',
+                selectedStatus.isCancelled(),
+                taskWithEditedStatusApplied,
+                'cancelled',
+            );
```

Comment:
It's a shame that these two lines wrap differently, so that the lines have to be read, rather than the pattern of the parameters being clear at a glance.

Reply:
Indeed, I thought about that too, but couldn't find a solution as good as yours